doc/hardware/peripherals/can/shell.rst

```diff
@@ -0,0 +1,266 @@
+hexadecimal format along with an optional CAN ID mask, also in hexadecimal format, for setting which
+bits in the CAN ID are to be matched. Refer to the interactive help output for this subcommand for
+further details on the supported arguments.
+
+.. code-block:: console
+
+   uart:~$ can filter add can@0 010
+   adding filter with standard (11-bit) CAN ID 0x010, CAN ID mask 0x7ff, data frames 1, RTR frames 0, CAN FD frames 0
+   filter ID: 0
+
+The filter ID (0 in the example above) returned is to be used when removing the CAN RX filter.
+
+Received CAN frames matching the added filter(s) are printed to the shell. A few examples are shown below:
+
+.. code-block:: console
+
+   # Flags  ID   Size  Data bytes
+   --       010   [8]  01 02 03 04 05 06 07 08
+   B-       010  [08]  01 02 03 04 05 06 07 08
+   BP       010  [03]  01 aa bb
+   --  00000010   [0]
+   --       010   [1]  20
+   --       010   [8]  remote transmission request
+
+The columns have the following meaning:
+
+* Flags
+
+  * ``B``: The frame has the CAN FD Baud Rate Switch (BRS) flag set.
+  * ``P``: The frame has the CAN FD Error State Indicator (ESI) flag set. The transmitting node is
+    in error-passive state.
+  * ``-``: Unset flag.
+
+* ID
+
+  * ``010``: The standard (11-bit) CAN ID of the frame in hexadecimal format, here 10h.
+  * ``00000010``: The extended (29-bit) CAN ID of the frame in hexadecimal format, here 10h.
+
+* Size
+
+  * ``[8]``: The number of frame data bytes in decimal format, here a classic CAN frame with 8 data
+    bytes.
+  * ``[08]``: The number of frame data bytes in decimal format, here a CAN FD frame with 8 data
+    bytes.
+
+* Data bytes
+
+  * ``01 02 03 04 05 06 07 08``: The frame data bytes in hexadecimal format, here the numbers from 1
+    through 8.
+  * ``remote transmission request``: The frame is a Remote Transmission Request (RTR) frame and thus
+    carries no data bytes.
+
+.. tip::
+   If :kconfig:option:`CONFIG_CAN_RX_TIMESTAMP` is enabled, each line will be prepended with a
+   timestamp from the free-running timestamp counter in the CAN controller.
+
+Configured CAN RX filters can be removed again using the ``can filter remove`` subcommand as shown
+below. The filter ID is the ID returned by the ``can filter add`` subcommand (0 in the example
+below).
+
+.. code-block:: console
+
+   uart:~$ can filter remove can@0 0
+   removing filter with ID 0
+
+Sending
+*******
+
+CAN frames can be queued for transmission using the ``can send`` subcommand as shown below. The
+subcommand accepts a CAN ID in hexadecimal format and optionally a number of data bytes, also
+specified in hexadecimal. Refer to the interactive help output for this subcommand for further
+details on the supported arguments.
+
+.. code-block:: console
+
+   uart:~$ can send can@0 010 1 2 3 4 5 6 7 8
+   enqueuing CAN frame #2 with standard (11-bit) CAN ID 0x010, RTR 0, CAN FD 0, BRS 0, DLC 8
+   CAN frame #0 successfully sent
```

Comment:
should this be `frame #2`? mentioning it just in case as this might be a copy-paste issue or a problem elsewhere?

Reply:
Nice catch. Fixed.